kivy/uix/behaviors/focus.py

```diff
@@ -375,7 +375,10 @@ def __init__(self, **kwargs):
         fbind('focus_previous', self._set_on_focus_previous)
 
     def _on_focusable(self, instance, value):
-        if self.disabled or not self.is_focusable:
+        if not value and self.focus:
+            self.focus = False
+            self._unbind_keyboard()
+        elif self.disabled or not self.is_focusable:
             self.focus = False
 
     def _on_focus(self, instance, value, *largs):
```

Comment:
Ideally, setting `self.focus = False` should be enough as that should release everything in `_on_focus`. The problem is that if you set first `is_focusable` to False, `_on_focus` will not do anything because of the `if self.is_focusable and self.keyboard_mode == 'auto':`. 

I think a better fix would then be to change 

```py
if self.is_focusable and self.keyboard_mode == 'auto':
     if value:
         self._bind_keyboard()
     else:
        self._unbind_keyboard()
```
to
```py
if self.keyboard_mode == 'auto':
     if not value:
         self._unbind_keyboard()
     elif self.is_focusable:
        self._bind_keyboard()
```